Algorithm.CSharp/ETFConstituentUniverseRSIAlphaModelAlgorithm.cs

```diff
@@ -0,0 +1,213 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Alphas;
+using QuantConnect.Algorithm.Framework.Execution;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using QuantConnect.Data;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Example algorithm demonstrating the usage of the RSI indicator
+    /// in combination with ETF constituents data to replicate the weighting
+    /// of the ETF's assets in our own account.
+    /// </summary>
+    public class ETFConstituentUniverseRSIAlphaModelAlgorithm : QCAlgorithm
+    {
+        /// <summary>
+        /// Store constituents data when doing universe selection for later use in the alpha model
+        /// </summary>
+        public List<ETFConstituentData> Constituents = new List<ETFConstituentData>();
+        
+        /// <summary>
+        /// Initialize the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2020, 12, 1);
+            SetEndDate(2021, 1, 31);
+            SetCash(100000);
+            
+            SetAlpha(new ConstituentWeightedRsiAlphaModel());
+            SetPortfolioConstruction(new InsightWeightingPortfolioConstructionModel());
+            SetExecution(new ImmediateExecutionModel());
+
+            var spy = AddEquity("SPY", Resolution.Hour).Symbol;
+            
+            // We load hourly data for ETF constituents in this algorithm
+            UniverseSettings.Resolution = Resolution.Hour;
+            AddUniverse(Universe.ETF(spy, UniverseSettings, FilterETFConstituents));
+        }
+
+        /// <summary>
+        /// Filters ETF constituents and adds the resulting Symbols to the ETF constituent universe
+        /// </summary>
+        /// <param name="constituents">ETF constituents, i.e. the components of the ETF and their weighting</param>
+        /// <returns>Symbols to add to universe</returns>
+        public IEnumerable<Symbol> FilterETFConstituents(IEnumerable<ETFConstituentData> constituents)
+        {
+            Constituents = constituents
+                .Where(x => x.Weight != null && x.Weight >= 0.001m)
+                .ToList();
+
+            return Constituents
+                .Select(x => x.Symbol)
+                .ToList();
+        }
+
+        /// <summary>
+        /// no-op
+        /// </summary>
+        public override void OnData(Slice data)
+        {
+        }
```

Comment:
Don't think this is required?
`.ToList();` in `return Constituents.Select(x => x.Symbol).ToList();` not required either?
The cleaner & simpler the better 👍